src/ArrayHelper.php

```diff
@@ -1423,23 +1465,21 @@ private static function parseMixedPath(array|float|int|string $path, string $del
     {
         if (is_array($path)) {
             $newPath = [];
+
             foreach ($path as $key) {
                 if (is_string($key)) {
-                    $parsedPath = StringHelper::parsePath($key, $delimiter);
-                    $newPath = array_merge($newPath, $parsedPath);
+                    $newPath[] = StringHelper::parsePath($key, $delimiter);
                     continue;
                 }
 
                 if (is_array($key)) {
-                    /** @var list<float|int|string> $parsedPath */
-                    $parsedPath = self::parseMixedPath($key, $delimiter);
-                    $newPath = array_merge($newPath, $parsedPath);
+                    $newPath[] = self::parseMixedPath($key, $delimiter);
                     continue;
                 }
 
-                $newPath[] = $key;
+                $newPath[] = [$key];
             }
-            return $newPath;
+            return array_merge(...$newPath);
```

Comment:
It doesn't give much performance unless key is very-very deep. Doesn't make it worse though.